tests/clients/test_alerts.py

```diff
@@ -92,3 +258,101 @@ def test_alerts_client_calls_get_aggregate_data_with_expected_value_and_param(
         alert_client = AlertsClient(mock_alerts_service, mock_alert_rules_service)
         alert_client.get_aggregate_data("alert-id")
         mock_alerts_service.get_aggregate_data.assert_called_once_with("alert-id")
+
+    def test_alerts_client_get_all_alert_details_calls_get_details_for_each_page(
+        self, mock_alerts_service_with_pages, mock_alert_rules_service
+    ):
+        mock_alerts_service = mock_alerts_service_with_pages(ascending=True)
+        alert_client = AlertsClient(mock_alerts_service, mock_alert_rules_service)
+        query = AlertQuery()
+        list(alert_client.get_all_alert_details(query))
+        assert mock_alerts_service.get_details.call_args_list[0][0][0] == [
+            "A",
+            "B",
+            "C",
+        ]
+        assert mock_alerts_service.get_details.call_args_list[1][0][0] == [
+            "D",
+            "E",
+            "F",
+        ]
+
+    @pytest.mark.parametrize(
+        "sort_key",
+        [
+            "AlertId",
+            "TenantId",
```

Comment:
Seems to weird to be able to sort by tenant ID, wouldn't that just always be the same value?

Reply:
yep, but it's a possible sort key so I figure we should at least make sure we handle it correctly if a user attempts it.